pkg/remoteconfig/state/repository.go

```diff
@@ -329,10 +335,17 @@ func (r *Repository) CurrentState() (RepositoryState, error) {
 	var configs []ConfigState
 	var cached []CachedFile
 
-	for path, metadata := range r.metadata {
-		configs = append(configs, configStateFromMetadata(metadata))
-		cached = append(cached, cachedFileFromMetadata(path, metadata))
-	}
+	r.metadata.Range(func(path, value any) bool {
+		metadata, ok := value.(Metadata)
+		if ok {
+			configs = append(configs, configStateFromMetadata(metadata))
+			cached = append(cached, cachedFileFromMetadata(path.(string), metadata))
+		} else {
+			// Log the error but continue processing the rest of the configs
+			log.Printf("Failed to convert metadata for %s", path)
```

Comment:
What's the error level? Can't we use `Warnf`?

Reply:
We have to use "log" package so it's either Print, Fatal or Panic